core/src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -4936,6 +4932,109 @@ abstract class MergeIntoSuiteBase
     expectErrorContains = "Cannot add column 'value.a.z' with type 'void'",
     expectErrorWithoutEvolutionContains = "All nested columns must match")
 
+  testNestedStructsEvolution("new source column in map struct")(
```

Comment:
Add tests where the map contains the same attributes in the target and source with different types, e.g. int vs. string:

```
target = """{ "key": "A", "map": { "key": { "a": 1 } } }"""
source ="""{ "key": "A", "map": { "key": { "a": "2" } } }"""
```